packages/jitm/src/class-post-connection-jitm.php

```diff
@@ -0,0 +1,505 @@
+				jQuery( document ).ready( function( $ ) {
+					$( '#jp-switch-connection-owner' ).on( 'submit', function( e ) {
+						var formData = $( this ).serialize();
+						var submitBtn = document.getElementById( 'jp-switch-connection-owner-submit' );
+						var results = document.getElementById( 'jp-switch-user-results' );
+
+						submitBtn.disabled = true;
+
+						$.ajax( {
+							type        : "POST",
+							url         : "<?php echo esc_url( get_rest_url() . 'jetpack/v4/connection/owner' ); ?>",
+							data        : formData,
+							headers     : {
+								'X-WP-Nonce': "<?php echo esc_js( wp_create_nonce( 'wp_rest' ) ); ?>",
+							},
+							success: function() {
+								results.innerHTML = "<?php esc_html_e( 'Success!', 'jetpack' ); ?>";
+								setTimeout( function() {
+									$( '#jetpack-notice-switch-connection-owner' ).hide( 'slow' );
+								}, 1000 );
+							}
+						} ).done( function() {
+							submitBtn.disabled = false;
+						} );
+
+						e.preventDefault();
+						return false;
+					} );
+				} );
+			</script>
+			<?php
+		} else {
+			echo '<p>' . esc_html__( 'Every Jetpack site needs at least one connected admin for the features to work properly. Please connect to your WordPress.com account via the button below. Once you connect, you may refresh this page to see an option to change the connection owner.', 'jetpack' ) . '</p>';
+			$connect_url = \Jetpack::init()->build_connect_url( false, false, 'delete_connection_owner_notice' );
+			echo "<a href='" . esc_url( $connect_url ) . "' target='_blank' rel='noopener noreferrer' class='button-primary'>" . esc_html__( 'Connect to WordPress.com', 'jetpack' ) . '</a>';
+		}
+
+		echo '<p>';
+		printf(
+			wp_kses(
+				/* translators: URL to Jetpack support doc regarding the primary user. */
+				__( "<a href='%s' target='_blank' rel='noopener noreferrer'>Learn more</a> about the connection owner and what will break if you do not have one.", 'jetpack' ),
+				array(
+					'a' => array(
+						'href'   => true,
+						'target' => true,
+						'rel'    => true,
+					),
+				)
+			),
+			esc_url( Redirect::get_url( 'jetpack-support-primary-user' ) )
+		);
+		echo '</p>';
+		echo '<p>';
+		printf(
+			wp_kses(
+				/* translators: URL to contact Jetpack support. */
+				__( 'As always, feel free to <a href="%s" target="_blank" rel="noopener noreferrer">contact our support team</a> if you have any questions.', 'jetpack' ),
+				array(
+					'a' => array(
+						'href'   => true,
+						'target' => true,
+						'rel'    => true,
+					),
+				)
+			),
+			esc_url( Redirect::get_url( 'jetpack-contact-support' ) )
+		);
+		echo '</p>';
+		echo '</div>';
+	}
+
+	/**
+	 * Dismisses a JITM feature class so that it will no longer be shown.
+	 *
+	 * @param string $id The id of the JITM that was dismissed.
+	 * @param string $feature_class The feature class of the JITM that was dismissed.
+	 *
+	 * @return bool Always true.
+	 */
+	public function dismiss( $id, $feature_class ) {
```

Comment:
There's a lot of overlap with this post-connection dismiss method and the pre-connection dismiss method. Is there a way that we can minimize duplication?

Reply:
Removed duplication.